sw-sysemu/mmu.cpp

```diff
@@ -150,677 +77,6 @@ static inline void check_store_breakpoint(const Hart& cpu, uint64_t addr)
-
-        if (mprot & MPROT_ENABLE_SECURE_MEMORY) {
-            if (paddr_is_dram_mcode(addr)) {
-                if (!spio)
-                    throw_access_fault(vaddr, macc);
-            }
-            else if (paddr_is_dram_mdata(addr)) {
-                if (!spio && (effective_execution_mode(cpu, macc) != Privilege::M))
-                    throw_access_fault(vaddr, macc);
-            }
-            else if (paddr_is_dram_scode(addr)) {
-                if (!spio && (effective_execution_mode(cpu, macc) != Privilege::M))
-                    throw_access_fault(vaddr, macc);
-            }
-            else if (paddr_is_dram_osbox(addr)) {
-                if (!spio && (mprot & MPROT_DISABLE_OSBOX_ACCESS))
-                    throw_access_fault(vaddr, macc);
-            }
-            else if (addr >= pma_dram_limit(spio, mprot)) {
-                throw_access_fault(vaddr, macc);
-            }
-        } else {
-            if (paddr_is_dram_mbox(addr)) {
-                if (!spio && (effective_execution_mode(cpu, macc) != Privilege::M))
-                    throw_access_fault(vaddr, macc);
-            }
-            else if (paddr_is_dram_sbox(addr)) {
-                if (!spio && (mprot & MPROT_DISABLE_OSBOX_ACCESS))
-                    throw_access_fault(vaddr, macc);
-            }
-            else if (paddr_is_dram_osbox(addr)) {
-                if (!spio && (mprot & MPROT_DISABLE_OSBOX_ACCESS))
-                    throw_access_fault(vaddr, macc);
-            }
-            else if (addr >= pma_dram_limit(spio, mprot)) {
-                throw_access_fault(vaddr, macc);
-            }
-        }
-        return truncated_dram_addr(cpu, addr);
-    }
-
-    if (paddr_is_sp_rom(addr) || paddr_is_sp_sram(addr)) {
-        if (!spio || paddr_is_sp_sram_code(addr))
-            throw_access_fault(vaddr, macc);
-        return addr;
-    }
-
-    throw_access_fault(vaddr, macc);
-}
-#else // !EMU_HAS_PMA
-static uint64_t pma_check_data_access(const Hart& cpu, uint64_t vaddr,
-                                      uint64_t addr, size_t size,
-                                      mem_access_type macc,
-                                      mreg_t mask = mreg_t(-1),
-                                      cacheop_type cop = CacheOp_None)
-{
-    (void)cpu;
-    (void)vaddr;
-    (void)size;
-    (void)macc;
-    (void)mask;
-    (void)cop;
-    return addr;
-}
-
-static uint64_t pma_check_ptw_access(const Hart& cpu, uint64_t vaddr,
-                                     uint64_t addr, mem_access_type macc)
-{
-    (void)cpu;
-    (void)vaddr;
-    (void)macc;
-    return addr;
-}
-
-static uint64_t pma_check_fetch_access(const Hart& cpu, uint64_t vaddr,
-                                       uint64_t addr, size_t size)
-{
-    (void)cpu;
-    (void)vaddr;
-    (void)size;
-    return addr;
-}
-
-#endif // EMU_HAS_PMA
-
-
-static uint64_t vmemtranslate(const Hart& cpu, uint64_t vaddr, size_t size,
```

Comment:
No. This is the *core* of MMU. You can't move it outside.

Now, we'll have to possibly tell people that at least in ETSOC1 and ERBIUM setting SATP to anything than BARE MODE is a big-no-no, but this code must stay here.